pkg/config/storage.go

```diff
@@ -1,72 +1,84 @@
-	storage := ComponentStorageSettings{
-		Name: name,
-		Size: size,
-		Path: path,
+// GetStorage gets the storage with the given name
+func (lci *LocalConfigInfo) GetStorage(storageName string) *localConfigProvider.LocalStorage {
+	for _, storage := range lci.ListStorage() {
+		if storageName == storage.Name {
+			return &storage
+		}
 	}
+	return nil
+}
+
+// CreateStorage sets the storage related information in the local configuration
+func (lci *LocalConfigInfo) CreateStorage(storage localConfigProvider.LocalStorage) error {
 	err := lci.SetConfiguration("storage", storage)
 	if err != nil {
-		return ComponentStorageSettings{}, err
+		return err
 	}
-	return storage, err
+	return err
 }
 
-func (lci *LocalConfigInfo) StorageExists(storageName string) bool {
-	for _, storage := range lci.GetStorage() {
-		if storageName == storage.Name {
-			return true
-		}
+// ListStorage gets all the storage from the config
+func (lci *LocalConfigInfo) ListStorage() []localConfigProvider.LocalStorage {
+	if lci.componentSettings.Storage == nil {
+		return []localConfigProvider.LocalStorage{}
 	}
-	return false
-}
 
-func (lci *LocalConfigInfo) StorageList() ([]ComponentStorageSettings, error) {
-	storageConfigList := lci.GetStorage()
-	var storageList []ComponentStorageSettings
-	for _, storage := range storageConfigList {
-		storageList = append(storageList, ComponentStorageSettings{
+	var storageList []localConfigProvider.LocalStorage
+	for _, storage := range *lci.componentSettings.Storage {
+		storageList = append(storageList, localConfigProvider.LocalStorage{
 			Name: storage.Name,
 			Path: storage.Path,
 			Size: storage.Size,
 		})
 	}
-	return storageList, nil
+	return storageList
 }
 
-func (lci *LocalConfigInfo) ValidateStorage(storageName, storagePath string) error {
-	for _, storage := range lci.GetStorage() {
-		if storage.Name == storageName {
-			return errors.Errorf("there already is a storage with the name %s", storageName)
-		}
-		if storage.Path == storagePath {
-			return errors.Errorf("there already is a storage mounted at %s", storagePath)
-		}
+// DeleteStorage deletes the storage with the given name
+func (lci *LocalConfigInfo) DeleteStorage(name string) error {
+	storage := lci.GetStorage(name)
+	if storage == nil {
+		return errors.Errorf("storage named %s doesn't exists", name)
 	}
-	return nil
+	return lci.DeleteFromConfigurationList("storage", name)
 }
 
-func (lci *LocalConfigInfo) StorageDelete(name string) error {
-	exists := lci.StorageExists(name)
-	if !exists {
-		return errors.Errorf("storage named %s doesn't exists", name)
+// CompleteStorage completes the given storage
+func (lci *LocalConfigInfo) CompleteStorage(storage *localConfigProvider.LocalStorage) {}
+
+// ValidateStorage validates the given storage
+func (lci *LocalConfigInfo) ValidateStorage(storage localConfigProvider.LocalStorage) error {
+	if storage.Size == "" || storage.Path == "" {
```

Comment:
Can we add a comment here that this error is expected only when working with s2i components and not with devfile components?

Reply:
This is in the `localConfig` package used by s2i components and we should not mention devfile here.